From 607e186d070df2bc8c5abafbc949e8ef568614bd Mon Sep 17 00:00:00 2001
From: jasper-smit-servicenow <jasper.smit@servicenow.com>
Date: Thu, 18 Jul 2024 09:45:22 +0200
Subject: [PATCH 085/125] Update X509V3_get_d2i.pod returned pointer needs to
 be freed

CLA: trivial

Reviewed-by: Tomas Mraz <tomas@openssl.org>
Reviewed-by: Tom Cosgrove <tom.cosgrove@arm.com>
Reviewed-by: Todd Short <todd.short@me.com>
(Merged from https://github.com/openssl/openssl/pull/24927)

(cherry picked from commit a4fd94851261c55f9ad020bf22d4f29bda0b58be)
(cherry picked from commit 12c38af865a0a60c98f6b63de5be4b8ce2d1ace5)
---
 doc/man3/X509V3_get_d2i.pod | 1 +
 1 file changed, 1 insertion(+)

diff --git a/doc/man3/X509V3_get_d2i.pod b/doc/man3/X509V3_get_d2i.pod
index 4a2e81b0db..dc00090b84 100644
--- a/doc/man3/X509V3_get_d2i.pod
+++ b/doc/man3/X509V3_get_d2i.pod
@@ -108,6 +108,7 @@ The function X509V3_get_d2i() and its variants
 will return NULL if the extension is not
 found, occurs multiple times or cannot be decoded. It is possible to
 determine the precise reason by checking the value of I<*crit>.
+The returned pointer must be explicitly freed.
 
 The function X509V3_add1_i2d() and its variants allocate B<X509_EXTENSION>
 objects on STACK I<*x> depending on I<flags>. The B<X509_EXTENSION> objects
-- 
2.39.5 (Apple Git-154)

